account.go

```diff
@@ -0,0 +1,356 @@
+			// equal gas price as that of this transaction.
+			if strings.Compare(err.Error(), ErrIncorrectNonce.Error()) == 0 {
+				return err
+			}
+			log.Println(err)
+			continue
+		}
+
+		// If post-condition check passes, proceed to wait for a specified
+		// number of blocks to be confirmed after the transaction's block
+		if postConditionCheck == nil || postConditionCheck() {
+			break
+		}
+
+		// Wait for sometime before attempting to execute the transaction
+		// again. If context is done, return error to indicate that
+		// post-condition failed
+		select {
+		case <-ctx.Done():
+			return ErrPostConditionCheckFailed
+		case <-time.After(sleepDurationMs * time.Millisecond):
+
+		}
+
+		// Increase delay for next round but saturate at 30s
+		sleepDurationMs = time.Duration(float64(sleepDurationMs) * 1.6)
+		if sleepDurationMs > 30000 {
+			sleepDurationMs = 30000
+		}
+	}
+
+	// At this point the transaction executed successfully and the
+	// post-condition check has passed. This means that we can now proceed to
+	// wait for a pre-defined number of blocks to be confirmed on the
+	// blockchain after the transaction's block is confirmed
+
+	// Attempt to get block number of transaction. If context times out, an
+	// error will be returned.
+	blockNumber, err := account.client.TxBlockNumber(ctx, txHash.String())
+	if err != nil {
+		return err
+	}
+
+	// Attempt to get current block number. If context times out, an error will
+	// be returned.
+	currentBlockNumber, err := account.client.CurrentBlockNumber(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Keep getting current block number until it is greater than
+	// 'waitForBlocks' + transaction's block number. If context times out, the
+	// error is returned.
+	for big.NewInt(0).Sub(currentBlockNumber, blockNumber).Cmp(big.NewInt(waitForBlocks)) < 0 {
+		currentBlockNumber, err = account.client.CurrentBlockNumber(ctx)
+		if err != nil {
+			select {
+			case <-ctx.Done():
+				return ctx.Err()
+			case <-time.After(5 * time.Millisecond):
+			}
+			continue
+		}
+	}
+	return nil
+}
+
+// Transfer transfers eth from the account to an ethereum address.
+func (account *account) Transfer(ctx context.Context, to common.Address, value *big.Int, confirmBlocks int64) error {
+
+	// Pre-condition check: Check if the account has enough balance
+	preConditionCheck := func() bool {
+		accountBalance, err := account.client.BalanceOf(ctx, account.Address(), &account.callOpts)
+		return err == nil && accountBalance.Cmp(value) >= 0
+	}
+
+	// Transaction: Transfer eth to address
+	f := func(transactOpts bind.TransactOpts) (*types.Transaction, error) {
+		bound := bind.NewBoundContract(to, abi.ABI{}, nil, account.client.EthClient, nil)
+
+		transactor := &bind.TransactOpts{
+			From:     transactOpts.From,
+			Nonce:    transactOpts.Nonce,
+			Signer:   transactOpts.Signer,
+			Value:    value,
+			GasPrice: transactOpts.GasPrice,
+			GasLimit: 21000,
+			Context:  transactOpts.Context,
```

Comment:
I am fairly sure that `Context: transactOpts` should be `Context: ctx`.

Reply:
Corrected and pushed.